src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,349 @@
+import com.openlattice.edm.EntitySet
+import com.openlattice.edm.type.EntityType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.hazelcast.HazelcastQueue
+import com.openlattice.retrofit.RhizomeByteConverterFactory
+import com.openlattice.retrofit.RhizomeCallAdapterFactory
+import com.openlattice.retrofit.RhizomeJacksonConverterFactory
+import com.openlattice.rhizome.proxy.RetrofitBuilders
+import com.openlattice.hazelcast.processors.shuttle.UpdateIntegrationEntryProcessor
+import com.openlattice.shuttle.logs.Blackbox
+import com.openlattice.hazelcast.mapstores.shuttle.INTEGRATION_STATUS
+import com.openlattice.shuttle.payload.JdbcPayload
+import com.openlattice.shuttle.payload.Payload
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import okhttp3.FormBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import retrofit2.Retrofit
+import java.io.IOException
+import java.net.URL
+import java.util.*
+import java.util.concurrent.Executors
+import java.util.concurrent.Semaphore
+
+private val logger = LoggerFactory.getLogger(IntegrationService::class.java)
+private const val fetchSize = 10_000
+private const val readRateLimit = 1_000
+private const val uploadBatchSize = 10_000
+private val threadCount = 2 * Runtime.getRuntime().availableProcessors()
+
+private lateinit var logEntityType: EntityType
+private lateinit var httpClient: OkHttpClient
+
+@Service
+class IntegrationService(
+        private val hazelcastInstance: HazelcastInstance,
+        private val missionParameters: MissionParameters,
+        private val idService: EntityKeyIdService,
+        private val entitySetManager: EntitySetManager,
+        private val reservationService: HazelcastAclKeyReservationService,
+        private val blackbox: Blackbox
+) {
+
+    private val integrations = HazelcastMap.INTEGRATIONS.getMap( hazelcastInstance )
+    private val entitySets = HazelcastMap.ENTITY_SETS.getMap( hazelcastInstance )
+    private val entityTypes = HazelcastMap.ENTITY_TYPES.getMap( hazelcastInstance )
+    private val propertyTypes = HazelcastMap.PROPERTY_TYPES.getMap( hazelcastInstance )
+    private val integrationJobs = HazelcastMap.INTEGRATION_JOBS.getMap( hazelcastInstance )
+    private val jobQueue = HazelcastQueue.INTEGRATION_JOBS.getQueue( hazelcastInstance )
+    private val semaphore = Semaphore(threadCount)
+    private val executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(threadCount))
+    private val statusPredicate = Predicates.or(
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.IN_PROGRESS),
+            Predicates.equal(INTEGRATION_STATUS, IntegrationStatus.QUEUED)
+    )
+    private val authAPI = AuthAPI(
+            missionParameters.auth.getProperty("apiDomain"),
```

Comment:
are these properties constantified anywhere?

Reply:
In the yaml but that's it